src/widget/wtracktableview.cpp

```diff
@@ -424,163 +298,10 @@ void WTrackTableView::loadTrackModel(QAbstractItemModel *model) {
-    connect(m_pClearOutroCueAction, SIGNAL(triggered()),
-            this, SLOT(slotClearOutroCue()));
-
-    m_pClearLoopAction = new QAction(tr("Loop"), this);
-    connect(m_pClearLoopAction, SIGNAL(triggered()),
-            this, SLOT(slotClearLoop()));
-
-    m_pClearKeyAction = new QAction(tr("Key"), this);
-    connect(m_pClearKeyAction, SIGNAL(triggered()),
-            this, SLOT(slotClearKey()));
-
-    m_pClearReplayGainAction = new QAction(tr("ReplayGain"), this);
-    connect(m_pClearReplayGainAction, SIGNAL(triggered()),
-            this, SLOT(slotClearReplayGain()));
-
-    m_pClearWaveformAction = new QAction(tr("Waveform"), this);
-    connect(m_pClearWaveformAction, SIGNAL(triggered()),
-            this, SLOT(slotClearWaveform()));
-
-    m_pClearAllMetadataAction = new QAction(tr("All"), this);
-    connect(m_pClearAllMetadataAction, SIGNAL(triggered()),
-            this, SLOT(slotClearAllMetadata()));
-
-
-    m_pBpmLockAction = new QAction(tr("Lock BPM"), this);
-    m_pBpmUnlockAction = new QAction(tr("Unlock BPM"), this);
-    connect(m_pBpmLockAction, SIGNAL(triggered()),
-            this, SLOT(slotLockBpm()));
-    connect(m_pBpmUnlockAction, SIGNAL(triggered()),
-            this, SLOT(slotUnlockBpm()));
-
-    //BPM edit actions
-    m_pBpmDoubleAction = new QAction(tr("Double BPM"), this);
-    m_pBpmHalveAction = new QAction(tr("Halve BPM"), this);
-    m_pBpmTwoThirdsAction = new QAction(tr("2/3 BPM"), this);
-    m_pBpmThreeFourthsAction = new QAction(tr("3/4 BPM"), this);
-    m_pBpmFourThirdsAction = new QAction(tr("4/3 BPM"), this);
-    m_pBpmThreeHalvesAction = new QAction(tr("3/2 BPM"), this);
-
-    connect(m_pBpmDoubleAction, &QAction::triggered,
-            this, [this] { slotScaleBpm(Beats::DOUBLE); });
-    connect(m_pBpmHalveAction, &QAction::triggered,
-            this, [this] { slotScaleBpm(Beats::HALVE); });
-    connect(m_pBpmTwoThirdsAction, &QAction::triggered,
-            this, [this] { slotScaleBpm(Beats::TWOTHIRDS); });
-    connect(m_pBpmThreeFourthsAction, &QAction::triggered,
-            this, [this] { slotScaleBpm(Beats::THREEFOURTHS); });
-    connect(m_pBpmFourThirdsAction, &QAction::triggered,
-            this, [this] { slotScaleBpm(Beats::FOURTHIRDS); });
-    connect(m_pBpmThreeHalvesAction, &QAction::triggered,
-            this, [this] { slotScaleBpm(Beats::THREEHALVES); });
-
-    ColorPaletteSettings colorPaletteSettings(m_pConfig);
-    m_pColorPickerAction = new WColorPickerAction(WColorPicker::Option::AllowNoColor, colorPaletteSettings.getTrackColorPalette(), this);
-    m_pColorPickerAction->setObjectName("TrackColorPickerAction");
-    connect(m_pColorPickerAction,
-            &WColorPickerAction::colorPicked,
-            this,
-            &WTrackTableView::slotColorPicked);
+    // Initialize m_pMenu only after track model has been loaded.
+    m_pMenu = new WTrackMenu(this, m_pConfig, m_pTrackCollectionManager, WTrackMenu::Filter::All, trackModel);
```

Comment:
A new WTrackMenu is allocated here, but the old one was never deleted.

Reply:
A new allocation will be made if loadTrackModel is called more than once. In either case, I'll use `unique_ptr` here.

Reply:
Please don't use std::unique_ptr for parented Qt objects. Use parented_ptr/make_parented instead to indicate this relationship.

Reply:
Even though we need to delete it separately from the Qt object tree?

Reply:
Thinking about it further, it is inefficient to completely destroy and recreate WTrackMenu every time the TableModel* changes. Now that the code has been cleaned up a bit and WTrackMenu isn't as tightly coupled to TableModel*, it may be easier to refactor. We could go back to having WTrackMenu created in the WTrackTableView constructor and remove the TrackModel* argument from the constructor again as well as remove the Flags from the constructor. Then we could add a `setFlagsAndTableModel` method to WTrackMenu to call in `WTrackTableView::loadTrackModel`. This way we can check the flags passed are compatible with the TableModel* without coupling that to the construction of WTrackMenu. The WTrackMenu constructor could unconditionally create all the QActions and QMenus and show/hide them in `setFlagsAndTableModel` instead of creating/destroying (use [QWidget::removeAction](https://doc.qt.io/qt-5/qwidget.html#removeAction)).

Reply:
... or maybe that would be premature optimization and we should just fix the memory leak currently in WTrackTableView.

Reply:
I had strived to optimize memory usage by WTrackMenu by making features optional. This means that a lot of QMenus and QActions will not be initialized if feature flags are limited.

WTrackTableView is used by the few library views like Tracks, AutoDJ, Playlist.
WTrackProperty is used by each deck and sampler. So if 64 sampler skin is activated, this means at least around 75 instances of WTrackMenu will be created. Not creating all the menus and actions unconditionally will save some memory.

Reply:
Does the TrackModel change dynamically during a session? As I can assume, a TrackModel is inherent property of the playlist, autodj list, track list, etc. and it's not supposed to change.

Reply:
> WTrackProperty is used by each deck and sampler. So if 64 sampler skin is activated, this means at least around 75 instances of WTrackMenu will be created. Not creating all the menus and actions unconditionally will save some memory.

That is a good point. Destroying and creating WTrackMenu when a different library view changes is a small price compared to optimizing a widget used on 64 samplers. So I guess leave it how it is, just fix the memory leak in WTrackTableView.